src/modules/cartPage.ts

```diff
@@ -0,0 +1,134 @@
+'use strict';
+import { OnButtonClick, OnProductClick } from './interfaces/customTypes';
+import { IProduct } from './interfaces/product.interface';
+
+type OnChangeAmount = (event: Event, id: number) => void;
+
+export class CartPage {
+    private parent: Element;
+    private cartDataLayout: string;
+    private productsData: IProduct[];
+    private selectedProducts: number[];
+    private onChangeAmount: OnChangeAmount;
+    private onProductClick: OnProductClick;
+    private onButtonClick: OnButtonClick;
+
+    constructor(
+        parent: Element,
+        productsData: IProduct[],
+        selectedProducts: number[],
+        onChangeAmount: OnChangeAmount,
+        onProductClick: OnProductClick,
+        onButtonClick: OnButtonClick
+    ) {
+        this.productsData = productsData;
+        this.parent = parent;
+        this.selectedProducts = selectedProducts;
+        this.onChangeAmount = onChangeAmount;
+        this.onProductClick = onProductClick;
+        this.onButtonClick = onButtonClick;
+
+        this.cartDataLayout = `
+            <div class="cart-page_wrapper">
+                <div class="selected-list">
+                    <h2 class="h2">Bag</h2>
+                </div>
+                <div class="summary">
+                    <h3 class="h3 summary__header"></h3>
+                    <input class="input promo_code" type="text" placeholder="Promo code">
+                    <p class="summary__total-price"></p>
+                    <button type="submit" class="button button_buy">Buy</button>
+                </div>
+            </div>
+            <div class="disabled-area hidden"></div>`;
+    }
+
+    public drawCartPage() {
+        this.parent.innerHTML = this.cartDataLayout;
+        const selectedList: Element = document.querySelector('.selected-list')!;
+
+        const summaryHeader: Element = document.querySelector('.summary__header')!;
+        const totalPriceContainer: Element = document.querySelector('.summary__total-price')!;
+        const inputPromoCode = document.querySelector('.promo_code')! as HTMLInputElement;
+        let totalAmount: number = 0;
+        let totalPrice: number = 0;
+        this.productsData.forEach((product) => {
+            if (this.selectedProducts.includes(product.id)) {
+                let amount = this.selectedProducts.reduce((amount, currentItem) => {
+                    if (currentItem === product.id) {
+                        amount += 1;
+                    }
+                    return amount;
+                }, 0);
```

Comment:
Never use same name for variable for outer scope and inner arrow function.

Reply:
Fixed.